deployment/cdk/opensearch-service-migration/lib/fetch-migration-stack.ts

```diff
@@ -0,0 +1,103 @@
+import {Secret as SMSecret} from "aws-cdk-lib/aws-secretsmanager";
+import {join} from "path";
+import {readFileSync} from "fs"
+import {StackPropsExt} from "./stack-composer";
+import {StringParameter} from "aws-cdk-lib/aws-ssm";
+
+export interface FetchMigrationProps extends StackPropsExt {
+    readonly vpc: IVpc,
+    readonly dpPipelineTemplatePath: string,
+    readonly sourceEndpoint: string
+}
+
+export class FetchMigrationStack extends Stack {
+
+    constructor(scope: Construct, id: string, props: FetchMigrationProps) {
+        super(scope, id, props);
+
+        // Import required values
+        const targetClusterEndpoint = StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/osClusterEndpoint`)
+        const domainAccessGroupId = StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/osAccessSecurityGroupId`)
+        // This SG allows outbound access for ECR access as well as communication with other services in the cluster
+        const serviceConnectGroupId = StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/serviceConnectSecurityGroupId`)
+
+        const ecsCluster = Cluster.fromClusterAttributes(this, 'ecsCluster', {
+            clusterName: `migration-${props.stage}-ecs-cluster`,
+            vpc: props.vpc
+        })
+
+        // ECS Task Definition
+        const fetchMigrationFargateTask = new FargateTaskDefinition(this, "fetchMigrationFargateTask", {
+            memoryLimitMiB: 2048,
+            cpu: 512
+        });
+
+        new StringParameter(this, 'SSMParameterFetchMigrationTaskDefArn', {
+            description: 'OpenSearch Migration Parameter for Fetch Migration task definition ARN',
+            parameterName: `/migration/${props.stage}/${props.defaultDeployId}/fetchMigrationTaskDefArn`,
+            stringValue: fetchMigrationFargateTask.taskDefinitionArn
+        });
+        new StringParameter(this, 'SSMParameterFetchMigrationTaskRoleArn', {
+            description: 'OpenSearch Migration Parameter for Fetch Migration task role ARN',
+            parameterName: `/migration/${props.stage}/${props.defaultDeployId}/fetchMigrationTaskRoleArn`,
+            stringValue: fetchMigrationFargateTask.taskRole.roleArn
+        });
+        new StringParameter(this, 'SSMParameterFetchMigrationTaskExecRoleArn', {
+            description: 'OpenSearch Migration Parameter for Fetch Migration task exec role ARN',
+            parameterName: `/migration/${props.stage}/${props.defaultDeployId}/fetchMigrationTaskExecRoleArn`,
+            stringValue: fetchMigrationFargateTask.obtainExecutionRole().roleArn
+        });
+
+        // Create Fetch Migration Container
+        const fetchMigrationContainer = fetchMigrationFargateTask.addContainer("fetchMigrationContainer", {
+            image: ContainerImage.fromAsset(join(__dirname, "../../../..", "FetchMigration")),
```

Comment:
This is less concerning for me but something I want to try out in the future as I think it could be beneficial: https://opensearch.atlassian.net/browse/MIGRATIONS-1420